atomic_reactor/tasks/binary_container_build.py

```diff
@@ -86,9 +92,68 @@ def execute(self) -> None:
             logger.info("Build finished succesfully! Pushing image to %s", dest_tag)
             podman_remote.push_container(dest_tag, registry_config=config.registry)
 
+    def acquire_remote_resource(self, remote_hosts_config: dict) -> remote_host.LockedResource:
+        """Lock a build slot on a remote host."""
+        logger.info("Acquiring a build slot on a remote host")
+        pool = remote_host.RemoteHostsPool.from_config(remote_hosts_config, self._params.platform)
+        resource = pool.lock_resource(prid=self._params.user_params["pipeline_run_name"])
+        if not resource:
+            raise BuildTaskError(
+                "Failed to acquire a build slot on any remote host! See the logs for more details."
+            )
+        return resource
+
+
+@functools.lru_cache
+def which_podman() -> str:
+    """Return the full path to the podman or podman-remote executable. Prefer podman."""
+    podman = shutil.which("podman") or shutil.which("podman-remote")
+    if not podman:
+        raise BuildTaskError("Could not find either podman or podman-remote in $PATH!")
+    return podman
+
 
 class PodmanRemote:
-    """Wrapper for running podman --remote commands on a remote host."""
+    """Wrapper for running podman --remote commands on a remote host.
+
+    Works with both podman-remote and full podman.
+    """
+
+    def __init__(self, connection_name: str):
+        """Initialize a PodmanRemote instance.
+
+        :param connection_name: the name of an existing podman-remote connection
+        """
+        self._connection_name = connection_name
+
+    @property
+    def _podman_remote_cmd(self) -> List[str]:
+        return [which_podman(), "--remote", f"--connection={self._connection_name}"]
+
+    @classmethod
+    def setup_for(cls, host: remote_host.RemoteHost):
+        """Set up a connection for the specified remote host and return a PodmanRemote instance."""
+        connection_name = host.hostname
```

Comment:
can we rather use pipeline run name for connection name?

Reply:
updated